File: 1/golang/main.go
package main

import (
	"database/sql"
	"encoding/csv"
	"fmt"
	_ "github.com/go-sql-driver/mysql"
	//"github.com/gosuri/uilive"
	"os"
	"time"
)

const host = "mysql"
const user = "my"
const pass = "my"
const database = "golang"

const table = "10e5-row"

const filePath = "/app/resources/" + table + ".csv"

func main() {
	defer dbTruncate()

	start := time.Now()
	defer func() {
		duration := time.Since(start)
		fmt.Println(duration.Microseconds(), "µs")
	}()

	//fmt.Println("Start!")
	//defer fmt.Println("Finish!")

	// connect mysql
	db := connectMysql(host, user, pass, database)
	defer db.Close()

	// open file
	file := openFile(filePath)
	defer file.Close()

	//writer := uilive.New()
	//writer.Start()
	//defer writer.Stop()

	for i, value := range readCsvFile(file) {
		if i == 0 {
			continue
		}
		dbInsert(db, table, value)
		//fmt.Fprintf(writer, "Row line - %d \n", i)
	}

}

func connectMysql(host, user, pass, database string) *sql.DB {
	db, err := sql.Open("mysql", user+":"+pass+"@tcp("+host+")/"+database)
	if err != nil {
		panic(err.Error())
	}
	return db
}

func dbInsert(db *sql.DB, table string, value []string) {
	_, err := db.Query("INSERT INTO `"+table+"`  (uid, manufacturer_part_number, manufacturer, quantity) VALUES (?, ?, ?, ?)", value[0], value[2], value[3], value[4])
	if err != nil {
		panic(err.Error())
	}
}

func dbTruncate() {
	db := connectMysql(host, user, pass, database)

	_, err := db.Query("TRUNCATE `" + table + "`")
	if err != nil {
		panic(err.Error())
	}
}

func openFile(path string) *os.File {
	file, err := os.Open(path)
	if err != nil {
		panic(err.Error())
	}
	return file
}

func readCsvFile(file *os.File) [][]string {
	csvReader := csv.NewReader(file)
	records, err := csvReader.ReadAll()
	if err != nil {
		panic(err.Error())
	}
	return records
}

File: 1/php/main.php
<?php

const host = "mysql";
const user = "my";
const pass = "my";
const database = "php";
const table = "10e6-row";
const filePath = "/app/resources/" . table . ".csv";


function main()
{
    dbTruncate();

    $start = microtime(true);

    $conn = connectMysql(host, user, pass, database);

    $file = openFile(filePath);


    $row = 1;
    while (($data = fgetcsv($file)) !== FALSE) {
        if ($row == 1) {
            $row++;
            continue;
        }
        dbInsert($conn, table, $data);
        $row++;
    }


    fclose($file);

    $conn = null;

    $execution_time = (microtime(true) - $start) * 1000000;
    echo floor($execution_time) . " µs\n";
}

function connectMysql(string $host, string $user, string $pass, string $database): PDO
{
    return new PDO("mysql:host=$host;dbname=$database;charset=UTF8", $user, $pass);
}

function dbInsert(PDO $conn, $table, $value)
{
    $sql = "INSERT INTO `$table` (`uid`, `manufacturer_part_number`, `manufacturer`, `quantity`) VALUES ('$value[0]', '$value[2]', '$value[3]', '$value[4]')";

    $conn->query($sql);
}

function dbTruncate()
{
    $conn = connectMysql(host, user, pass, database);
    if (!$conn->query("TRUNCATE `" . table . "`")) {
        die("Error: " . $conn->error . PHP_EOL);
    }
}

function openFile($path)
{
    $file = fopen($path, "r");
    if ($file === false) {
        die("Open file!" . PHP_EOL);
    }
    return $file;
}


main();